src/Psalm/Internal/Type/AssertionReconciler.php

```diff
@@ -1257,21 +1179,15 @@ private static function handleLiteralEqualityWithString(
                     return $existing_var_type;
                 }
 
-                if ($scalar_type === 'class-string'
-                    || $scalar_type === 'interface-string'
-                    || $scalar_type === 'trait-string'
-                ) {
+                if ($assertion_type instanceof TLiteralClassString) {
```

Comment:
this condition can be dropped (there are a few like that in the file)

Reply:
Why can it be dropped?

Reply:
Done!

Reply:
The one on line 1456 can still be dropped :)